### PR TITLE
Set FluentBit self logs to buffer in memory instead of filesystem

### PR DESCRIPTION
**Pull request by avilevy18**
Parameterized `storage.type` config with variable `BufferInMemory`. Edit component to always use in memory buffering for self logs

Following: http://b/226668416, this PR changes self logs to buffer in memory as opposed to the filesystem.

Soak Test:
image

### CODE REVIEW

**Review by qingling128 — APPROVED** [review]
LGTM

**Comment by qingling128** [comment]
SLES 12 was failing, but it's not related to this PR.

File: confgenerator/logging_receivers.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ type LoggingReceiverFilesMixin struct {
 	ExcludePaths            []string        `yaml:"exclude_paths,omitempty"`
 	WildcardRefreshInterval *time.Duration  `yaml:"wildcard_refresh_interval,omitempty" validate:"omitempty,min=1s,multipleof_time=1s"`
 	MultilineRules          []MultilineRule `yaml:"-"`
+	BufferInMemory          bool
```
Comment (quentinmit) [inline confgenerator/logging_receivers.go:60]: Mark it as `yaml:"-"` so it can't be set by user config.

**Review by quentinmit — COMMENTED** [review]
Is there a doc somewhere I can read on why we want to do this?

(Also, presumably fluent-bit is still writing state somewhere on disk so it doesn't reprocess the same log lines, right?)

File: confgenerator/logging_receivers.go (hunk 1 above)
Comment (avilevy18) [reply confgenerator/logging_receivers.go]: Changed in: [f67580f](https://github.com/GoogleCloudPlatform/ops-agent/pull/631/commits/f67580f4f76efa0303dc9986a7867855b006bd27)

**Pull request merged**